### DISCUSSION TITLE
How to convert soft line breaks to paragraph breaks

### DISCUSSION THREAD

**Discussion by vnugent** [facebook/lexical#3663] [Q&A]
Hi, I'm looking for tips on how to write a plugin that converts soft line breaks (shift enter key) to paragraph breaks as users type or paste text.

Basically, I want the output to be: 
```
p
   text: one
p 
   text: break
p
   text: two
```

Playground
<img width="470" alt="Screen Shot 2023-01-01 at 9 25 33 PM" src="https://user-images.githubusercontent.com/3805254/210183784-ae638c1a-ac80-4833-b121-220c0c407d2d.png">

**Comment by moy2010** [comment]
You could replace the `LineBreakNode` with your own node, and override the `createDOM` method accordingly.
